extensions/stackdriver/log/logger.h

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *constexpr int kDefaultTimeoutMillisecond = 10000;
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "extensions/common/context.h"
+#include "extensions/stackdriver/log/exporter.h"
+#include "google/logging/v2/logging.pb.h"
+
+namespace Extensions {
+namespace Stackdriver {
+namespace Log {
+
+#ifdef NULL_PLUGIN
+using Envoy::Extensions::Common::Wasm::Null::Plugin::Extensions::Stackdriver::
+    Log::Exporter;
+#endif
+
+// Size of an entries.write request is 5 Mb:
+// https://cloud.google.com/logging/quotas.
+constexpr int kMaxRotationBytesSize = 4000000;
```

Comment:
aren't constexpr not suggested to be used in header files?

Reply:
Removed.